src/commands/up.ts

```diff
@@ -1,51 +1,75 @@
 import cliSpinners from 'cli-spinners';
 import { existsSync } from 'std/fs/exists.ts';
 import * as path from 'std/path/mod.ts';
+import { animals, uniqueNamesGenerator } from 'unique-names-generator';
 import winston, { Logger } from 'winston';
 import { CloudGraph } from '../cloud-graph/index.ts';
-import { parseEnvironment } from '../environments/index.ts';
+import { DatacenterRecord } from '../datacenters/store.ts';
+import { Environment, parseEnvironment } from '../environments/index.ts';
 import { ImageRepository } from '../oci/index.ts';
 import { Pipeline, PlanContextLevel } from '../pipeline/index.ts';
 import { BaseCommand, CommandHelper, GlobalOptions } from './base-command.ts';
 import { destroyEnvironment } from './destroy/environment.ts';
 import { streamLogs } from './logs.ts';
 
 type UpOptions = GlobalOptions & {
-  datacenter: string;
-  environment: string;
   verbose: boolean;
   debug: boolean;
   ingress?: string[];
-};
+} & ({ environment: string; datacenter?: never } | { datacenter: string; environment?: never });
 
 const UpCommand = BaseCommand()
   .description('Spin up an environment that will clean itself up when you terminate the process')
   .arguments('[...components:string]')
-  .option('-d, --datacenter <datacenter:string>', 'The datacenter to use for the environment', { required: true })
-  .option('-e, --environment <environment:string>', 'The name of your tmp environment', { default: 'local' })
+  .option('-d, --datacenter <datacenter:string>', 'The datacenter to use for the environment', {
+    required: true,
+    conflicts: ['environment'],
+  })
+  .option('-e, --environment <environment:string>', 'The name of your tmp environment', {
+    required: true,
+    conflicts: ['datacenter'],
+  })
   .option('-i, --ingress <ingress:string>', 'Mappings of ingress rules for this component to subdomains', {
     collect: true,
   })
   .option('-v, --verbose [verbose:boolean]', 'Turn on verbose logs', { default: false })
   .option('--debug [debug:boolean]', 'Deploy component in debug mode', { default: true })
-  .action(up_action);
+  .action(up_action as any);
```

Comment:
Why do we need the `as any` here. Worries me we are breaking typechecking.

Reply:
Typescript has a lot of difficulty with XOR like functionality, and it looks like cliffy's type inference for the action is running into the same issues where it is not correctly figuring out the typing for environment/datacenter. I suppose I could change this and do the conflicts checks in code, but it seems like a missed opportunity since cliffy seems to handle it all well.